app/models/patient.rb

```diff
@@ -8,4 +8,15 @@ class Patient
 	has_many :pregnancies
 
   validates_presence_of :name, :primary_phone
+
+  def self.search(name_or_phone_str) # TODO optimize
+    name_matches = Patient.where name: name_or_phone_str
+    primary_matches = Patient.where primary_phone: name_or_phone_str
+    secondary_matches = Patient.where secondary_phone: name_or_phone_str
+    results = []
```

Comment:
It makes a lot of sense to move this logic to the model. Good call! 

The only suggestion I have is that we avoid looping through the results array in the model. We could take out lines 17-21 and simply define results on line 16 ```results = name_matches | primary_matches | secondary_matches```